packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/secretValuesWalkthrough.ts

```diff
@@ -176,12 +176,20 @@ const enterSecretName = async (invalidNames: string[]) =>
 
 const secretValueDefaultMessage = (secretName: string) => `Enter the value for ${secretName}:`;
 
+export const secretValueValidator = () => (input?: string) => {
+  if (typeof input !== 'string' || input.length === 0 || input.length > 2048) {
+    return 'Secret value must be between 1 and 2048 characters long';
+  }
+  return true;
+};
+
 const enterSecretValue = async (message: string) =>
   (
     await inquirer.prompt<{ secretValue: string }>({
       type: 'password',
       name: 'secretValue',
       message,
+      validate: secretValueValidator(),
```

Comment:
This looks good - I don't think you need to invoked `secretValueValidator` here. I think you can pass in the function without invoking it and removing the top level function from `secretValueValidator`.